File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.cpp
class Solution {
public:
int removeDuplicates(vector<int>& nums) {
    if(nums.empty()) return 0;
    if(nums.begin() == nums.end()) return 1;
    map<int,int> unique;
    for (auto it = nums.begin(); it != nums.end() ; it++) {
        unique[*it] = 1;
    }
    int i = 0;
    for (auto it = unique.begin(); it != unique.end(); ++it) {
        nums[i] = it->first;
        i++;
    }
    return unique.size();
}
};

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) == 1:
            return
        lastDigit = -101
        currIndex = 0
        i = 0
        while i < len(nums):
            if nums[i] == lastDigit:
                nums[i:] = nums[i+1:]
            else:
                i += 1
            lastDigit = nums[i-1]
            

    


        